src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -0,0 +1,136 @@
+package uk.ac.cam.cl.dtg.isaac.api;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import jakarta.ws.rs.core.Response;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import uk.ac.cam.cl.dtg.segue.api.UsersFacade;
+import uk.ac.cam.cl.dtg.segue.api.monitors.RegistrationMisuseHandler;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.stream.Stream;
+
+import static org.easymock.EasyMock.createNiceMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.niceMock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.ANONYMOUS_USER;
+
+public class UsersFacadeIT extends IsaacIntegrationJupiterTest {
+    private UsersFacade usersFacade;
+
+    @BeforeAll
+    public static void beforeAll() {
+        misuseMonitor.registerHandler(RegistrationMisuseHandler.class.getSimpleName(),
+                new RegistrationMisuseHandler(emailManager, properties));
+    }
+
+    @BeforeEach
+    public void beforeEach() throws Exception {
+        misuseMonitor.resetMisuseCount("0.0.0.0", RegistrationMisuseHandler.class.getSimpleName());
+        this.usersFacade = new UsersFacade(properties, userAccountManager, logManager, userAssociationManager, misuseMonitor, userPreferenceManager, schoolListReader);
+    }
+
+    @ParameterizedTest
+    @MethodSource("validRegistrationParameters")
+    public void createUser_validRegistrationParameters(String email, String password, String dob, String familyName, String givenName) {
+        HttpSession mockSession = createNiceMock(HttpSession.class);
```

Comment:
Duplicated setup code. You could probably wrap all this in a method that just returns a response.